src/box/port.h

```diff
@@ -142,6 +142,65 @@ enum port_c_entry_type {
 	PORT_C_ENTRY_BOOL,
 	PORT_C_ENTRY_MP,
 	PORT_C_ENTRY_MP_OBJECT,
+	PORT_C_ENTRY_ITERABLE,
+};
+
+struct port_c_iterator;
+
+/**
+ * Type of function that can be used as an iterator_create method in port_c.
+ * It is called only when the port is still alive.
+ */
+typedef void
+(*port_c_iterator_create_f)(void *data, struct port_c_iterator *it);
+
+/**
+ * Type of function that can be used as an iterator_next method in port_c.
+ * The iterator is passed as the first argument.
+ * The port to yield values is passed as the second argument.
+ * The third argument is a pointer to a flag which is set when the iterator
+ * is exhausted.
+ * Function must return 0 in the case of success. If eof is not reached,
+ * port `out` must be initialized, otherwise it mustn't.
+ * Also, the function can be called even after eof was reached - in this
+ * case it must work correctly and consistently return eof.
+ * In the case of error, port out must not be initialized, diag must be set
+ * and -1 must be returned.
+ */
+typedef int
+(*port_c_iterator_next_f)(struct port_c_iterator *it, struct port *out,
+			  bool *is_eof);
+
+/**
+ * An iterable object stored in port_c.
+ * When it is dumped, an iterator is created and is dumped instead of
+ * this object. Is dumped only when it is possible, for example, we cannot
+ * pack iterator to MsgPack, so it is dumped as MP_NIL there.
+ */
+struct port_c_iterable {
+	/**
+	 * Creates an iterator.
+	 * Mustn't be NULL.
+	 */
+	port_c_iterator_create_f iterator_create;
+	/** The iterable object itself. */
+	void *data;
```

Comment:
Please write somewhere a note about the objects lifetime. It should be clear that:
- Iterable object (data) must stay alive until port is destroyed.
- An iterator should correctly handle a situation when the iterable object is destroyed from under its feet.

Reply:
Described the lifetime in `struct port_c_iterable` description.